src/backend/drm/common/fallback.rs

```diff
@@ -0,0 +1,347 @@
+//!
+//! Types to make fallback device initialization easier
+//!
```

Comment:
This module is quite a lot of boilerplate... Though I'm not really clear how this could be improved.

Reply:
I have tried to skip a lot of code by using macros.

I bet I could safe some more, but I wanted it to be a reasonable time investment then just optimizing for code size. If this code ever gets touched at all, it likely gets replaced, so I did not want to spend too much time with it.

The only thing that might be changed is adding some more trait implementations and the macros will help with that as well.

But it is not single purpose code at least. I want to re-use it for `EGLStreamDevice` vs `EGLDevice`, when nvidia support lands.